app/components/channel_modal.js

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {Component, PropTypes} from 'react';
+import {Animated, Dimensions} from 'react-native';
+
+const AnimatedView = Animated.View;
+const {width: deviceWidth, height: deviceHeight} = Dimensions.get('window');
+
+export default class ChannelModel extends Component {
+    static propTypes = {
+        children: PropTypes.node,
+        duration: PropTypes.number,
+        topOffset: PropTypes.number,
+        theme: React.PropTypes.object.isRequired,
+        visible: PropTypes.bool.isRequired
+    }
+
+    static defaultProps = {
+        duration: 400,
```

Comment:
Works for me. What filename should we use? Sounds like we could run into this for a lot of different views so do we want to create an all-encompassing file for these types of properties?